src/panels/developer-tools/state/developer-tools-state.js

```diff
@@ -297,8 +303,17 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
         return hass.states[key];
       })
       .filter(function (value) {
-        if (!value.entity_id.includes(_entityFilter.toLowerCase())) {
-          return false;
+        // If we have a match for the entity ID, we do not have to look further
+        const entityFilter = _entityFilter.toLowerCase();
+        if (!value.entity_id.includes(entityFilter)) {
+          // Entity ID does not match => check the friendly_name => if there is none,
+          // this entity is not relevant for rendering => return false
+          if (
+            value.attributes.friendly_name === undefined ||
```

Comment:
Why are we no longer rendering entities that don't have a `friendly_name`?

Reply:
There was condition missing, that I now added: If there is no entity filter set, then we can skip this whole section since we will render all entities. If there is a filter set, we check the entity ID first (that we do regardless of friendly name). 

If the entity ID does however not match the filter, we now still have to check the friendly name, but of course only if there actually is one. If not, we know that the entity is already irrelevant based on the current filter.